dotNETStandard/HealthVaultProviderManagementPortal/HealthVaultProviderManagementPortal/Controllers/GoalsController.cs

```diff
@@ -30,26 +30,32 @@ public class GoalsController : Controller
         /// </summary>
         public async Task<ActionResult> Index(Guid personId, Guid recordId)
         {
-            var response = await ExecuteMicrosoftHealthVaultRestApiAsync(api => api.GetGoalsAsync(), personId, recordId);
+            var response = await ExecuteMicrosoftHealthVaultRestApiAsync(api => api.Goals.GetActiveAsync(), personId, recordId);
             return View(response);
         }
 
         /// <summary>
         /// Get a goal for the logged in user
         /// </summary>
         [HttpGet]
-        public async Task<ActionResult> Goal(Guid id, Guid personId, Guid recordId)
+        public async Task<ActionResult> Goal(Guid? id, Guid personId, Guid recordId)
```

Comment:
>? [](start = 49, length = 1)

Show we have two different controllers depending on if the id is provided or not? This logic is kind of weird to follow. Consider having:
public async Task<ActionResult> Goal(Guid id, Guid personId, Guid recordId) and public async Task<ActionResult> Goal(Guid personId, Guid recordId) and separate the two methods.